### PR TITLE
Yet another set of hooks :P

### PR DESCRIPTION
**Pull request by emanuele45**

### CODE REVIEW

**Comment by live627** [comment]
Why not  copy how the query constructs are done in Search.php and put them all into variables instead of hardcoding them into the query?

**Comment by emanuele45** [comment]
I thought about that possibility (or something similar), but in the end all the infos that are "hard-coded" in the query are mandatory. SMF needs them to proceed, no way around it.
The search query has to be flexible because it can retrieve different things, but here what SMF wants is that set of data at least. And if someone wants to not retrieve some of the informations he has no other ways than do several edits. So it becomes useless (I think) to pass an array of data that we already know it cannot be changed without break a lot of other things.

The idea behind this hooks (at least the one I had when I wrote them) is that modders usually (I think) want to retrieve something more (for example a new field they added) when posts are displayed.

Did I miss anything obvious? (that is something rather frequent :P)

Now that I read again my code I'm a bit unsure about the $msg_tables (can't say why though), and the fact that another hook could be useful for the query that retrieves the list of messages (i.e. populates $messages)...

File: Sources/Display.php [hunk 1]
```diff
@@ -1261,6 +1275,8 @@ function prepareDisplayContext($reset = false)
 	// Is this user the message author?
 	$output['is_message_author'] = $message['id_member'] == $user_info['id'];
 
+	call_integration_hook('integrate_prepare_context', array(&$output, &$message));
```
Comment (live627) [inline Sources/Display.php]: Could this be integrate_prepare_message_context, for clarity?

Comment (emanuele45) [inline Sources/Display.php]: Yep, you are right.

Comment (emanuele45) [inline Sources/Display.php]: What do you think about integrate_prepare_display_context so that is consistent with the name of the function?

**Comment by live627** [comment]
> The idea behind this hooks (at least the one I had when I wrote them) is that modders usually (I think) want to retrieve something more (for example a new field they added) when posts are displayed.

Well, yes, the ones in ths commit will be used for additions. In fact, most hooks are usedd to extend, although sometimes (rarely) they modify oor even remove/

> Did I miss anything obvious? (that is something rather frequent :P)

It'll be easier for me to comment on the diff for that :D

> another hook could be useful for the query that retrieves the list of messages (i.e. populates $messages)...

Yes, only, the one I am thinking of doesn't modify the main query (that can only goo so far without hampering general performance). 

You see, attachments are fetch separately. Which is a good thing. I;'d like to see a hook added there. Ah, screw it. I'll do it myself.

File: Sources/Display.php (hunk 1 above)
Comment (live627) [inline Sources/Display.php]: That'll work, too.

**Comment by jdarwood007** [comment]
I see no issues with this commit.  Merge it.

**Pull request merged**